portfolio/src/main/java/com/google/sps/servlets/HandleCommentsServlet.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/*
+ *  Servlet that adds comments
+ */
+@WebServlet("/handle-comments")
+public class HandleCommentsServlet extends HttpServlet {
+
+  @Override
+  public void init() {}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("add-content");
+    String color = request.getParameter("color");
+    String font_size = request.getParameter("font_size");
+    long time = System.currentTimeMillis();
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      String urlToRedirectToAfterUserLogsIn = "/handle-comments";
+      response.sendRedirect(userService.createLoginURL(urlToRedirectToAfterUserLogsIn));
```

Comment:
This servlet has no `doGet` implementation, so this redirection fails! I think this should either (1) re-issue the same post request after the user is logs in, or (2) redirect to the other servlet to display the comments.

If you do (2) though, the user has to type their comment again after logging in.